packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -254,8 +254,34 @@ class WebViewController {
     return _channel.invokeMethod("reload");
   }
 
-  Future<void> _updateSettings(Map<String, dynamic> update) async {
-    return _channel.invokeMethod('updateSettings', update);
+  Future<void> _updateSettings(_WebSettings update) async {
+    final Map<String, dynamic> updateMap = _settings.updatesMap(update);
+    _settings = update;
+    return _channel.invokeMethod('updateSettings', updateMap);
+  }
+
+/// Evaluates a JavaScript expression in the context of the current page.
+///
+/// On Android returns the evaluation result as a JSON formatted string.
+///
+/// On iOS depending on the value type the return value would be one of:
+///
+///  - For primitive JavaScript types: the value string formatted (e.g JavaScript 100 returns '100').
+///  - For JavaScript arrays of supported types: a string formatted NSArray(e.g '(1,2,3)').
+///  - Other non-primitive types are not supported on iOS and will complete the Future with an error.
+///
+/// The Future completes with an error if a JavaScript error occurred, or on iOS, if the type of the
+/// evaluated expression is not supported as described above.
+  Future<String> evaluateJavaScript(String jsString) async {
+    if (_settings.javaScriptMode == JavaScriptMode.disabled) {
+      throw Exception('JavaScript mode disabled');
```

Comment:
nit: throw a FlutterError here, the message should say what was wrong, e.g 'JavaScript mode must not be disables when calling evaluateJavascript').